### ISSUE TITLE
`exec user process caused: exec format error` with `agc logs engine`

### ISSUE THREAD

**Issue by multimeric** [aws/amazon-genomics-cli#211]
[//]: #  (Please do not include your AWS account or credentials as part of the bug request.)

**Describe the Bug**

I'm getting `standard_init_linux.go:228: exec user process caused: exec format error`

**Steps to Reproduce**

* Run a workflow using  `agc workflow run some_flow --context some_context`
* `agc logs engine --context some_context`

**Relevant Logs**

```
2021-12-07T14:06:06+11:00 𝒊  Showing engine logs for 'T2Context'
Tue, 07 Dec 2021 13:57:40 +1100 standard_init_linux.go:228: exec user process caused: exec format error
```

**Expected Behavior**

The logs to be shown.

**Actual Behavior**

The logs are not shown; instead I get this error.

Operating System:
AGC Version: 1.1.2
Was AGC setup with a custom bucket: No
Was AGC setup with a custom VPC: No

**Comment by multimeric** [comment]
Okay I have some more info. I looked in the actual cloudwatch log group for this (`NextflowEngineEngineLogGroup`), and this exact message appeared, which means that it's an actual error thrown by the engine and not by the `agc` as I mistakenly thought.

**Comment by multimeric** [comment]
Okay I've found another clue. When I look at the actual job definition of the job that's failing, it seems that the command is not correct, because it tries to run the actual zip file. This would explain why golang (ie Docker) is claiming this is an invalid executable. I have again replaced sensitive information with ALL_CAPS placeholders:
```bash
aws batch describe-jobs --jobs MY_JOB_ARN --query jobs[0].container.command 
[
    "s3://agc-MY-ACCOUNT-ID-MY-REGION-1/project/MY-PROJECT/userid/michaelrmi2UTCAG/context/MY-CONTEXT/workflow/MY-WORKFLOW/workflow.zip"
]
```

**Comment by multimeric** [comment]
I wonder if it relates to this: https://github.com/aws/amazon-genomics-cli/blob/e4ad17efa9a3724c7dd9bc2f37430c21ebcf9d4e/packages/cdk/lib/wes_adapter/amazon_genomics/wes/adapters/NextflowWESAdapter.py#L50-L52

**Comment by multimeric** [comment]
Okay it turns out I went in the wrong direction. I was able to eliminate some issues by running the `nf-core` examples, which work fine:

* The `zip` file is indeed the correct `cmd` for the jobs
* Also, the issue isn't the missing shebang line, since I added one to my workflow and it still doesn't work

I believe the difference comes from the way in which the workflow files are provided. In the `nf-core` examples the actual nextflow code is specified as a git URL:

https://github.com/aws/amazon-genomics-cli/blob/e4ad17efa9a3724c7dd9bc2f37430c21ebcf9d4e/examples/nf-core-project/workflows/atacseq/MANIFEST.json#L1-L7

I assume this means that the adapter is `git clone`ing the repository, which will correctly preserve the file permissions. However, my manifest file is:

```
{
  "mainWorkflowURL": "./main.nf",
  "inputFileURLs": [
    "params_aws.json"
  ],
  "engineOptions": "-resume"
}
```

So I'm using the directory option. It correctly zips the directory as I am able to download the bundle from S3. However notably, even though I have given my `main.nf` execute permissions, when I unzip the bundle, it no longer has these permissions, I think because .zip doesn't even preserve unix permissions. I'm wondering then if Batch is trying to run `main.nf` which does not have execute permissions, causing the error posted at the top of this thread.

**Comment by multimeric** [comment]
No, I guess it's not related to the file permissions either: I submitted a manually created zip file with the right permissions to the batch queue and that didn't help. Unless the container is unzipping it in a non permission-aware manner. But the linux `unzip` command does it correctly.

**Comment by multimeric** [comment]
Okay I finally worked it out! Due to #209 I wasn't able to use my preferred T instance types, so I foolishly chose the C6g series, which are ARM processors! So everything was configured correctly, but some part of the AGC stack was trying to run on ARM which resulted in this error. I've created a reproducible example here: https://github.com/multimeric/AgcArmError.

Although this is mostly user error, I think it would be helpful if the AGC could flag when the user accidentally or deliberately chooses an ARM processor, because this will fail and they may not realise.
